app/components/conferences/dayOfEvent/mainstage/videostreamer.js

```diff
@@ -41,50 +41,100 @@ export default function Videostreamer(props) {
-          <source src={props.src} type={props.type}></source>
-          <p className="vjs-no-js">
-            To view this video please enable JavaScript, and consider upgrading
-            to a web browser that
-            <a href="https://videojs.com/html5-video-support/" target="_blank">
-              supports HTML5 video
-            </a>
-          </p>
-        </video>
-        <Alert show={!ping} handleToast={handleToast} message={message} />
-      </Col>
+      {props.service === "local" && (
+        <>
+          <Head>
+            <link
+              href="https://vjs.zencdn.net/7.17.0/video-js.css"
+              rel="stylesheet"
+            />
+          </Head>
+          <Script
+            src="https://vjs.zencdn.net/7.17.0/video.min.js"
+            strategy="afterInteractive"
+            onLoad={() =>
+              console.log(
+                `script loaded correctly, window.FB has been populated`
+              )
+            }
+          />
+          <Col className={styles.video_root}>
+            <div className={styles.video_server}>
+              <Badge pill bg={"warning"} text={"black"}>
+                {props.region}
+              </Badge>
+            </div>
+            <video
+              autoPlay={true}
+              id={styles.my_video}
+              className="video-js vjs-big-play-centered vjs-responsive"
+              controls
+              preload="auto"
+              poster={props.poster}
+              data-setup="{}"
+              onError={handleError}
+              onProgress={handleProgress}
+              onLoadedData={handleLoad}
+              onWaiting={handleWait}
+              onStalled={handleStall}
+            >
+              <source src={props.src} type={props.type}></source>
+              <p className="vjs-no-js">
+                To view this video please enable JavaScript, and consider
+                upgrading to a web browser that
+                <a
+                  href="https://videojs.com/html5-video-support/"
+                  target="_blank"
+                >
+                  supports HTML5 video
+                </a>
+              </p>
+            </video>
+            <Alert show={!ping} handleToast={handleToast} message={message} />
+          </Col>
+        </>
+      )}
+      {props.service === "twitch" && (
+        <>
+          <div className="vidoestreamer-container" style={{ margin: 30 }}>
+            <div id="twitch-embed"></div>
+          </div>
+          <Script
+            src="https://embed.twitch.tv/embed/v1.js"
+            strategy={"afterInteractive"}
+            onLoad={() => {
+              console.log("succesfully loaded twitch.tv script");
+              const embed = new Twitch.Embed("twitch-embed", {
+                width: 1300,
+                height: 700,
+                channel: props.src,
+                layout: "video",
+                autoplay: false,
+                parent: ["127.0.0.1", "localhost"],
```

Comment:
Could you let me know what this does? Since if we are going to use this in production, it will definitely have a different IP.
If it is not necessary, please remove this, or if it is, try to find a way to handle the production and development server.
Thank you!

Reply:
From what I've read and understood from the Twitch docs, The `parent` key in the player object specifies the parent origin for embedding Twitch videos, and its value is an array of strings that lists the permitted IP addresses/Domains of the embedded content. It is necessary to add the `parent` key, otherwise the Twitch player won't show the livestream and throw some error related to not the right kind of environment to show livestream. In the production environment, we should replace the `parent` key's value with the actual array of domain, where the embedded Twitch livestream is expected to be displayed. Thus It's necessary to correctly set the parent value in-order to ensure that the embedded content is only displayed in the intended and trusted environment and to prevent any security vulnerabilities.

Reply:
Ok, thank you. Now I understand so it is like whitelisting the domain to show the Twitch stream.

Hmm, this is tricky, as you have seen we design things in RC4Conferences as a component, so we could not ask any user to edit a file before editing edit. Could you make this read from an environment variable? 
FYI, we have an env variable available during the build which is `NEXT_PUBLIC_API_URL=http://localhost:3000` (this is the URL where the NextJS client is running), so if you could use it in the parent key, that would be great. 

Thank you!